doc/DEVELOPERS.md

```diff
@@ -0,0 +1,256 @@
+
+If you see an error saying `pgmoneta: pgmoneta: Configuration not found: /etc/pgmoneta/pgmoneta.conf` running the above command. you may need to add the config file.
+
+    cd /etc
+    sudo mkdir pgmoneta
+    sudo vi pgmoneta.conf
+
+`pgmoneta.conf`'s content is
+
+    [pgmoneta]
+    host = *
+    metrics = 5001
+
+    base_dir = /home/pgmoneta
+
+    compression = zstd
+
+    storage_engine = local
+
+    retention = 7
+
+    log_type = file
+    log_level = info
+    log_path = /tmp/pgmoneta.log
+
+    unix_socket_dir = /tmp/
+
+    [primary]
+    host = localhost
+    port = 5432
+    user = repl
+
+In our main section called `[pgmoneta]` we setup `pgmoneta` to listen on all network addresses. We will enable Prometheus metrics on port 5001 and have the backups live in the `/home/pgmoneta` directory. All backups are being compressed with zstd and kept for 7 days. Logging will be performed at `info` level and put in a file called `/tmp/pgmoneta.log`. Last we specify the location of the `unix_socket_dir` used for management operations and the path for the PostgreSQL command line tools.
+
+Next we create a section called [primary] which has the information about our PostgreSQL instance. In this case it is running on localhost on port 5432 and we will use the repl user account to connect.
+
+Finally, you should be able to obtain the version of pgmoneta. Cheers!
+
+## Set up pgmoneta
+
+Let's give it a first try. The basic idea here is that we will use two user spaces of the OS: one is `postgres`, which will run PostgreSQL, and one is `pgmoneta`, which will run our pgmoneta and monitor PostgreSQL to backup.
+
+In many installations, there is also an operating system user named `postgres` that is used to run the PostgreSQL server. You can use the command
+
+    getent passwd | grep postgres
+
+to check if your OS has a user named postgres. if not use
+
+    sudo useradd -ms /bin/bash postgres
+    sudo passwd postgres
+
+If the postgres user already exists, don't forget to set its password for convenience.
+
+### 1. postgres
+
+Open a new window, switch to the `postgres` user. This section will always operate within this user space.
+
+    sudo su -
+    su - pgmoneta
+
+#### Initialize cluster
+
+If you use dnf to install your postgresql, chances are the binary file is in `/usr/bin/`
+
+    export PATH=/usr/usr/bin:$PATH
+    initdb /tmp/pgsql
+
+#### Remove default acess
+
+Remove last lines from `/tmp/pgsql/pg_hba.conf`
+
+    host    all             all             127.0.0.1/32            trust
+    host    all             all             ::1/128                 trust
+    host    replication     all             127.0.0.1/32            trust
+    host    replication     all             ::1/128                 trust
+
+#### Add access for users and a database
+
+Add new lines to `/tmp/pgsql/pg_hba.conf`
+
+    host    mydb             myuser          127.0.0.1/32            md5
```

Comment:
We should use `scram-sha-256` instead - and make sure that the default `postgresql.conf` file has that

Reply:
Here's a question regarding the default `postgresql.conf`:
`#password_encryption = scram-sha-256`
This code is commented out. Do we need to enable it by removing the #?

Reply:
So 12 and 13 has MD5 as the default, and 14+ is SCRAM-SHA-256, so it is best to make sure that it is explicit enabled for all versions